src/main/scala/rocket/CSR.scala

```diff
@@ -950,6 +950,7 @@ class CSRFile(
   io.gstatus.sd := io.gstatus.fs.andR || io.gstatus.xs.andR || io.gstatus.vs.andR
   io.gstatus.uxl := (if (usingUser) log2Ceil(xLen) - 4 else 0)
   io.gstatus.sd_rv32 := xLen == 32 && io.gstatus.sd
+  io.gstatus.xs := (if (usingRoCC) UInt(3) else UInt(0))
 
```

Comment:
Can't we just hardcode `reg_vsstatus.xs` to dirty?

Reply:
Yes, you're right. And make gstatus.xs won't help to fix the bug mentioned in https://github.com/chipsalliance/rocket-chip/pull/2983, since rocc_illegal use reg_vsstatus.xs.